test/e2e/allocator_test.go

```diff
@@ -76,7 +76,7 @@ func TestAllocator(t *testing.T) {
 		RequiredGameServerSelector:   &pb.LabelSelector{MatchLabels: map[string]string{agonesv1.FleetNameLabel: flt.ObjectMeta.Name}},
 		PreferredGameServerSelectors: []*pb.LabelSelector{{MatchLabels: map[string]string{agonesv1.FleetNameLabel: flt.ObjectMeta.Name}}},
 		Scheduling:                   pb.AllocationRequest_Packed,
-		MetaPatch:                    &pb.MetaPatch{Labels: map[string]string{"gslabel": "allocatedbytest"}},
+		Metadata:                     &pb.MetaPatch{Labels: map[string]string{"gslabel": "allocatedbytest"}},
```

Comment:
🤔 do we want to test both metadata and metapatch -- just to ensure the deprecated version continues to work?

Reply:
+1 also test for the input validation case when both MetaPatch and Metadata is set. You can do more rigorous testing in unit tests.

Reply:
Additional test cases for the metapatch and metadata field were added to `converter_test.go`